### PR TITLE
Fix --workspace_status_command empty value handling

### LINKED ISSUE #16037: --workspace_status_command: handling of empty values is surprising and inconsistent
### Description of the bug:

If the workspace status command prints a line containing a key and a trailing space, this line will be copied to `volatile-status.txt` (or `stable-statux.txt`). This is OK.

If the workspace status command prints a line containing a key only, this line gets discarded. This is somewhat surprising and undocumented.

If the workspace status command prints a line containing a key without value as the last line of the output, this line gets discarded, even if there is a a trailing space after the key. This is surprising, undocumented and inconsistent with example #1.

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

# Summary
My workspace status command:
```
#!/bin/bash
echo 'KEY_WITH_VALUE foo'
echo 'KEY_ONLY'
echo 'KEY_WITH_SPACE '
echo 'KEY_WITH_SPACE_AT_EOF '
```

My `bazel-out/volatile-status.txt`:
```
BUILD_TIMESTAMP 1659518612
KEY_WITH_SPACE
KEY_WITH_VALUE foo
```

# Details
```
diff --git a/.bazelversion b/.bazelversion
new file mode 100644
index 0000000..91ff572
--- /dev/null
+++ b/.bazelversion
@@ -0,0 +1 @@
+5.2.0
diff --git a/BUILD b/BUILD
new file mode 100644
index 0000000..f1fe42a
--- /dev/null
+++ b/BUILD
@@ -0,0 +1,12 @@
+genrule(
+  name = 'my-test-gen',
+  executable = True,
+  stamp = 1,
+  outs = ['my-test'],
+  cmd = """
+echo "#!/bin/bash" > $@
+echo "cat <<'EOT'" >> $@
+cat bazel-out/volatile-status.txt >> $@
+echo EOT >> $@
+"""
+)
diff --git a/WORKSPACE b/WORKSPACE
new file mode 100644
index 0000000..e69de29
diff --git a/subdir/workspace-status b/subdir/workspace-status
new file mode 100755
index 0000000..4d94172
--- /dev/null
+++ b/subdir/workspace-status
@@ -0,0 +1,5 @@
+#!/bin/bash
+echo 'KEY_WITH_VALUE foo'
+echo 'KEY_ONLY'
+echo 'KEY_WITH_SPACE '
+echo 'KEY_WITH_SPACE_AT_EOF '
```

```
$ bazelisk run --workspace_status_command subdir/workspace-status my-test
[...]
INFO: Build completed successfully, 2 total actions
BUILD_TIMESTAMP 1659519173
KEY_WITH_SPACE
KEY_WITH_VALUE foo
$
```
Same problem on release 4.2.2.

### Which operating system are you running Bazel on?

RHEL 7.9

### What is the output of `bazel info release`?

5.2.0

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.

_No response_

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?

_No response_

### Have you found anything relevant by searching the web?

I haven't found anything relevant, sorry, except the documentation (https://bazel.build/docs/user-manual#workspace-status):

> The program should print zero or more key/value pairs to standard output, one entry on each line, then exit with zero (otherwise the build fails). The key names can be anything but they may only use upper case letters and underscores. The first space after the key name separates it from the value. The value is the rest of the line (including additional whitespaces). Neither the key nor the value may span multiple lines. Keys must not be duplicated.

### Any other information, logs, or outputs that you want to share?

_No response_

### PR DESCRIPTION
**Pull request by wade-arista**
### Description
parseWorkspaceStatus() was silently discarding lines without a space due to input.trim() mangling trailing-space-at-EOF combined with the key-only check skipping them. parseValues() had the same key-only discard behavior.

Both now store an empty string for keys with no value, and skip genuinely empty lines. Unit tests added for both parsers covering key-only, trailing-space, and EOF edge cases.

### Motivation
Specific cases of interest:

1. Key only with no trailing space:

    echo "KEY_ONLY"
    echo "KEY_WITH_VALUE value"

2. Key only entry at EOF (trailing space optional):

    echo "KEY_WITH_VALUE value"
    echo "KEY_ONLY "

Fixes #16037

### Build API Changes
No

### Checklist

- [X] I have added tests for the new use cases (if any).
- [x] I have updated the documentation (if applicable).

### Release Notes

RELNOTES: None

### CODE REVIEW

**Comment by wade-arista** [comment]
@dabanki mind having a look?

**Comment by wade-arista** [comment]
@brentleyjones are you able to take a look at this? I just saw you triaged the issue it references.

**Comment by wade-arista** [comment]
@dabanki @brentleyjones bump

**Review by meisterT — APPROVED** [review]

**Comment by wade-arista** [comment]
Thanks @meisterT, would this be considered for a cherry-pick into 9?
